docs/docs/noir/standard_library/traits.md

```diff
@@ -52,6 +52,41 @@ impl<A, B, C, D, E> Default for (A, B, C, D, E)
 For primitive integer types, the return value of `default` is `0`. Container
 types such as arrays are filled with default values of their element type.
 
+
+## `std::convert`
+
+### `std::convert::From`
+
+#include_code from-trait noir_stdlib/src/convert.nr rust
+
+The `From` trait defines how to convert from a given type `T` to the type on which the trait is implemented.
+
+The Noir standard library provides a number of implementations of `From` between primitive types.
+#include_code from-impls noir_stdlib/src/convert.nr rust
+
+#### When to implement `From`
+
+As a general rule of thumb, `From` may be implemented in the [situations where it would be suitable in Rust](https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from):
+
+- The conversion is *infallible*: Noir does not provide an equivalent to Rust's `TryFrom`, if the conversion can fail then provide a named method instead.
+- The conversion is lossless: semantically, it should not lose or discard information. For example, `u32: From<u16>` can losslessly convert any `u16` into a valid `u32` such that the original `u16` can be recovered. On the other hand, `u16: From<u32>` should not be implemented as `2**16` is a `u32` which cannot be losslessly converted into a `u16`.
+- The conversion is *value-preserving*: the conceptual kind and meaning of the resulting value is the same, even though the Noir type and technical representation might be different. While it's possible to infallibly and losslessly convert a `u8` into a `str<2>` hex representation, `4u8` and `"04"` are too different for `str<2>: From<u8>` to be implemented.
+- The conversion is *obvious*: it's the only reasonable conversion between the two types. If there's ambiguity on how to convert between them such that. For instance rather than implementing `U128: From<[u8; 16]>`, the methods `U128::from_le_bytes` and `U128::from_be_bytes` are used as otherwise the endianness of the array would not be ambiguous, resulting in two potential values of `U128` from the same byte array.
+
+One additional recommendation specific to Noir is:
+- The conversion is *efficient*: it's relatively cheap to convert between the two types. Due to being a ZK DSL, it's more important to avoid unnecessary computation compared to Rust. If the implementation of `From` would encourage users to perform unnecessary conversions then this can result in additional proving time, it may be preferable to expose functionality such that this conversion may be avoided.
```

Comment:
```suggestion
- The conversion is *efficient*: it's relatively cheap to convert between the two types. Due to being a ZK DSL, it's more important to avoid unnecessary computation compared to Rust. If the implementation of `From` would encourage users to perform unnecessary conversion, resulting in additional proving time, then it may be preferable to expose functionality such that this conversion may be avoided.
```